test/models/flava/test_flava.py

```diff
@@ -155,50 +188,84 @@ def setUp(self):
             use_image_masking=True,
         )
 
-        mm_encoder = nn.Identity()
-        image_to_mm_projection = nn.Identity()
-        text_to_mm_projection = nn.Identity()
-        self.flava = FLAVAModel(
-            image_encoder=self.image_encoder,
-            text_encoder=self.text_encoder,
+    @pytest.fixture
+    def mm_encoder(self):
+        return nn.Identity()
+
+    @pytest.fixture
+    def image_to_mm_projection(self):
+        return nn.Identity()
+
+    @pytest.fixture
+    def text_to_mm_projection(self):
+        return nn.Identity()
+
+    @pytest.fixture
+    def text_projection(self):
+        return nn.Identity()
+
+    @pytest.fixture
+    def image_projection(self):
+        return nn.Identity()
+
+    @pytest.fixture
+    def flava(
+        self,
+        image_encoder,
+        text_encoder,
+        mm_encoder,
+        image_to_mm_projection,
+        text_to_mm_projection,
+        text_projection,
+        image_projection,
+    ):
+        return FLAVAModel(
```

Comment:
I think the behavior is the same in both cases. FLAVA sets dropout prob to 0.0 and uses layer norm, which is unaffected by eval mode. So we could add it but I don't think it would actually change anything

Reply:
I think as a good practice we should add it because the model may at some point introduce randomness. What do you think?

Reply:
I will just add it so it doesn't lead to any confusion in the future